lilypad/prompts.py

```diff
@@ -25,31 +28,234 @@
+    ) -> Sequence["ChatCompletionMessageParam"]: ...
+
+    @overload
+    def messages(self, provider: Literal["anthropic"]) -> Sequence["MessageParam"]: ...
+
+    @overload
+    def messages(self, provider: Literal["gemini"]) -> Sequence["ContentDict"]: ...
+
+    def messages(
+        self, provider: Literal["openai", "anthropic", "gemini"]
+    ) -> (
+        Sequence["ChatCompletionMessageParam"]
+        | Sequence["MessageParam"]
+        | Sequence["ContentDict"]
+    ):
+        """Return the messages array for the given provider converted from base."""
+        if provider == "openai":
+            from mirascope.core.openai._utils import convert_message_params
+
+            # type error needs resolution on mirascope side
+            return convert_message_params(self._base_message_params)  # pyright: ignore [reportArgumentType]
+        elif provider == "anthropic":
+            from mirascope.core.anthropic._utils import convert_message_params
+
+            # type error needs resolution on mirascope side
+            return convert_message_params(self._base_message_params)  # pyright: ignore [reportArgumentType]
+        elif provider == "gemini":
+            from mirascope.core.gemini._utils import convert_message_params
+
+            # type error needs resolution on mirascope side
+            return convert_message_params(self._base_message_params)  # pyright: ignore [reportArgumentType]
+        else:
+            raise NotImplementedError(f"Unknown provider: {provider}")
+
+    @overload
+    def call_params(self, provider: Literal["openai"]) -> "OpenAICallParams": ...
+
+    @overload
+    def call_params(self, provider: Literal["anthropic"]) -> "AnthropicCallParams": ...
+
+    @overload
+    def call_params(self, provider: Literal["gemini"]) -> "GeminiCallParams": ...
+
+    def call_params(
+        self, provider: Literal["openai", "anthropic", "gemini"]
+    ) -> "OpenAICallParams | AnthropicCallParams | GeminiCallParams":
+        """Return the call parameters for the given provider converted from common."""
+        if provider == "openai":
+            from mirascope.core.openai._utils._convert_common_call_params import (
+                convert_common_call_params,
+            )
+
+            return convert_common_call_params(self.common_call_params)
+        elif provider == "anthropic":
+            from mirascope.core.anthropic._utils._convert_common_call_params import (
+                convert_common_call_params,
+            )
+
+            return convert_common_call_params(self.common_call_params)
+        elif provider == "gemini":
+            from mirascope.core.gemini._utils._convert_common_call_params import (
+                convert_common_call_params,
+            )
+
+            return convert_common_call_params(self.common_call_params)
+        else:
+            raise NotImplementedError(f"Unknown provider: {provider}")
+
+
+def _construct_trace_attributes(
+    version: PromptPublic,
```

Comment:
rename prompt for variable?

Reply:
i used `version` everywhere (for both prompt and generation) since it's the version we're using. should i instead use `prompt` / `generation` or `prompt_version` / `generation_version`?

Reply:
i like prompt_version / generation_version a little better than version everywhere, even though its in the prompts file